cli/ss2

```diff
@@ -35,12 +38,82 @@ from pyroute2.netlink.diag import (SS_ESTABLISHED,
                                    SS_CONN)
 
 
-class Protocol(Callable):
+class UserCtxtMap(collections.Mapping):
+
+    _data = {}
+
+    _sk_inode_re = re.compile(r"socket:\[(?P<ino>\d+)\]")
+
+    _proc_sk_fd_cast = "/proc/%d/fd/%d"
+
+    def _parse_inode(self, pid, pconn):
+        sk_path = self._proc_sk_fd_cast % (pid, pconn.fd)
+        inode = None
+
+        sk_inode_raw = os.readlink(sk_path)
+        inode = self._sk_inode_re.search(sk_inode_raw).group('ino')
+
+        if not inode:
+            raise RuntimeError("Unexpected kernel sk inode outline")
+
+        return inode
+
+    def _enter_item(self, usr, pid, flow, ctxt):
+        sk_inode = int(self._parse_inode(pid, flow))
+        sk_fd = flow.fd
+
+        _enter_pid = lambda _: \
+            self._data[sk_inode][usr].__setitem__(pid, ctxt)
+
+        _enter_fd = lambda _: \
+            self._data[sk_inode][usr][pid]["fds"].append(sk_fd)
+
+        if sk_inode in self._data.keys():
+            if usr in self._data[sk_inode].keys():
+                if pid not in self._data[sk_inode][usr].keys():
+                    _enter_pid(ctxt)
+                _enter_fd(sk_fd)
+            else:
+                self._data[sk_inode][usr] = {}
+                _enter_pid(ctxt)
```

Comment:
Nevertheless, you use it here, and it's not only confusing, but also causes unnecessary variable transitions. Simply to ignore the argument in the function and use a variable with the same name from the upper context.